docs/en/ingest-management/elastic-agent/configuration/autodiscovery/kubernetes-conditions-autodiscover.asciidoc

```diff
@@ -0,0 +1,318 @@
+
+The `redis` input defined in the {agent} manifest only has `info` as metricset. You can find more metricsets in the {metricbeat-ref}/metricbeat-module-redis.html[Redis module page], as well as the configuration options.
+
+To deploy Redis, you can apply the following example manifest:
+
+[source,yaml]
+------------------------------------------------
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: test-redis-config
+data:
+  redis-config: |
+    maxmemory 2mb
+    maxmemory-policy allkeys-lru
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: redis
+  labels:
+    k8s-app: redis
+    app: redis
+spec:
+  containers:
+  - name: redis
+    image: redis:5.0.4
+    command:
+      - redis-server
+      - "/redis-master/redis.conf"
+    env:
+    - name: MASTER
+      value: "true"
+    ports:
+    - containerPort: 6379
+    resources:
+      limits:
+        cpu: "0.1"
+    volumeMounts:
+    - mountPath: /redis-master-data
+      name: data
+    - mountPath: /redis-master
+      name: config
+  volumes:
+    - name: data
+      emptyDir: {}
+    - name: config
+      configMap:
+        name: test-redis-config
+        items:
+        - key: redis-config
+          path: redis.conf
+------------------------------------------------
+
+You should now be able to see Redis data flowing in on index `metrics-redis.info-default`. Make sure that the port on your Redis manifest file and the one used in the Redis input match. 
+
+**Note:** All assets (dashboards, ingest pipelines, and so on) related to the Redis integration are not installed. You need to explicitly <<install-uninstall-integration-assets,install them through Kibana>>.
+
+== Example: Target pods by label
+
+WARNING: In some "As a Service" Kubernetes implementations, like GKE, the control plane nodes or even the pods running on them won’t be visible. In these cases, it won’t be possible to use scheduler metricsets, necessary for this example. Refer https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-kubernetes.html#_scheduler_and_controllermanager[here] to find more information.
+
+To set the target host dynamically only for a targeted Pod based on its labels, use a variable in the
+{agent} policy to return path information from the provider:
+
+[source,yaml]
+----
+- data_stream:
+    dataset: kubernetes.scheduler
+    type: metrics
+  metricsets:
+    - scheduler
+  bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
+  hosts:
+    - 'https://${kubernetes.pod.ip}:10259'
+  period: 10s
+  ssl.verification_mode: none
+  condition: ${kubernetes.labels.component} == 'kube-scheduler'
+----
+
+If you now deploy another Redis pod with a different port, it should be detected. You can check this by, for example, studying the field `service.address` under `metrics-redis.info-default`. It should now display 2 different services.
```

Comment:
How Redis is relevant here? Right before we are talking about `scheduler` 🤔 ?

Reply:
I wrote about that to give the user the chance to check how that works. I rephrased a little bit to make it more clear: "Following the Redis example, if you deploy another Redis Pod with a different port, it should be detected. To check this, go, for example, to the field `service.address` under `metrics-redis.info-default`. It should be displaying two different services.". But even if this doesn't make sense now, we can remove it. What do you think?